fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneScanAllEntriesTest.java

```diff
@@ -211,4 +226,260 @@ private void assertIndexEntryPrimaryKeyTuples(Set<Tuple> primaryKeys, RecordCurs
+            // For grouped records, structure is (group, recNo) or similar
+            // For non-synthetic parent records, the recNo is in the tuple
+            try (FDBRecordContext verifyContext = openContext()) {
+                FDBRecordStore verifyStore = dataModel.createOrOpenRecordStore(verifyContext);
+                FDBStoredRecord<?> storedRecord = verifyStore.loadRecord(primaryKey);
+                assertNotNull(storedRecord, "Record should exist");
+                com.google.protobuf.Message message = storedRecord.getRecord();
+                com.google.protobuf.Descriptors.FieldDescriptor recNoField =
+                        message.getDescriptorForType().findFieldByName("rec_no");
+                long recNo = (long) message.getField(recNoField);
+                assertEquals(0, recNo % 2, "All indexed records should have even recNo, but found: " + recNo);
+            }
+        }
+    }
+
+    @Nonnull
+    private FDBRecordStore.Builder getStoreBuilderFilterOddRecNo(@Nonnull FDBRecordContext context,
+                                                                   @Nonnull RecordMetaDataProvider metaData,
+                                                                   @Nonnull final KeySpacePath path) {
+        // Create an index maintenance filter that only indexes records with even recNo
+        IndexMaintenanceFilter evenRecNoFilter = (index, rec) -> {
+            com.google.protobuf.Descriptors.FieldDescriptor recNoField =
+                    rec.getDescriptorForType().findFieldByName("rec_no");
+            if (recNoField != null && rec.hasField(recNoField)) {
+                long recNo = (long) rec.getField(recNoField);
+                if (recNo % 2 == 0) {
+                    return IndexMaintenanceFilter.IndexValues.ALL;
+                }
+            }
+            return IndexMaintenanceFilter.IndexValues.NONE;
+        };
+
+        return getStoreBuilder(context, metaData, path)
+                .setIndexMaintenanceFilter(evenRecNoFilter);
+    }
+
+    @ParameterizedTest
+    @BooleanSource
+    public void indexScanWithRecNoIndexPredicateTest(boolean isGrouped) throws Exception {
+        final long seed = 5432198L;
+
+        // Create an index predicate that only indexes records with even recNo
+        // We'll create a predicate: rec_no % 2 == 0
+        // Since there's no modulo operator, we'll use an OR of specific even values
+        final Type.Record recordType = Type.Record.fromDescriptor(
+                TestRecordsGroupedParentChildProto.MyParentRecord.getDescriptor());
+        final QuantifiedObjectValue recordValue = QuantifiedObjectValue.of(Quantifier.current(), recordType);
+        final FieldValue recNoField = FieldValue.ofFieldName(recordValue, "rec_no");
+
+        // Create predicate for even rec_no values: recNo IN (1002, 1004, 1006, 1008, 1010)
+        final com.apple.foundationdb.record.query.plan.cascades.predicates.QueryPredicate evenRecNoPredicate =
```

Comment:
When I suggested filtering out even records, I intended for it to be an easy and semi-realistic filtering. This doesn't seem to work well for `Predicate`s, perhaps it would be better to add a `boolean` field to the parent record, that is `rec_no % 2` (or random) and filter on that value being true. 

This also bakes a fair amount of internals of `LuceneIndexTestDataModel` into the behavior of this test, which doesn't seem great.

Reply:
Changed to recNo > 1006, which requires a simple predicate. Is that sufficient?